kale/worker.py

```diff
@@ -222,7 +227,11 @@ def _cleanup_worker(self, signum, frame):
         """
         logger.info('Process sent signal %d. Cleaning up tasks...' % signum)
 
+        self._cleanup_lock.acquire()
+        self._terminated = True
```

Comment:
would this be better served with a context manager?

```
with self._cleanup_lock:
    self._terminated = True
    num_completed, num_incomplete = self._release_batch()
```

If there is an exception in `self._release_batch()`, the lock will get still get released.